### PR TITLE
Ranch ci proxy wrapper

### PR DESCRIPTION
**Pull request by swettk**

### CODE REVIEW

File: install-ci-tools.sh [hunk 1]
```diff
@@ -200,8 +200,46 @@ EOF
     ranch)
       if [ ! -x ranch ] || [ "$(./ranch version)" != "$version" ]; then
         rm -f ranch
-        curl -fsSL http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz | gunzip > ranch
+        curl -fsSL "http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz" | gunzip > ranch_real
+        chmod +x ranch_real
+      cat > ranch <<"EOF"
+#!/bin/bash
+set -e
```
Comment (bobzoller) [inline install-ci-tools.sh:207]: `set -euo pipefail`

File: install-ci-tools.sh [hunk 2]
```diff
@@ -200,8 +200,46 @@ EOF
...
+set -o pipefail
+
+# Make sure and clean up
+trap "exit" INT TERM ERR
+trap "kill 0" EXIT
```
Comment (bobzoller) [inline install-ci-tools.sh:212]: does `kill 0` clean up child processes?  can you explain these two lines a little more? ;)

File: install-ci-tools.sh [hunk 3]
```diff
@@ -200,8 +200,46 @@ EOF
...
+echo "$RANCH_PROXY_SSH_KEY" | base64 -d > .ssh_key
+chmod 600 .ssh_key
+
+sshcmd='ssh -o ExitOnForwardFailure=yes -i .ssh_key -l admin'
+case "$RANCH_ENDPOINT" in
+  *huevosbuenos.com*)
+    export RANCH_ENDPOINT="https://ranch-api-staging.internal.huevosbuenos.com:8005"
+    $sshcmd -L 8005:ranch-api-staging.internal.huevosbuenos.com:443 jump.us-east-1.dev-aws.goodeggs.com "sleep 900" &
+    ;;
+  *goodeggs.com*)
```
Comment (bobzoller) [inline install-ci-tools.sh:231]: the ranch cli defaults to ranch-api.goodeggs.com and does not require `RANCH_ENDPOINT` to be set, so probably make this `*)` or add a third default case?

File: install-ci-tools.sh [hunk 4]
```diff
@@ -200,8 +200,46 @@ EOF
...
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+export HOSTALIASES=${script_dir}/hostaliases
+echo "$RANCH_PROXY_SSH_KEY" | base64 -d > .ssh_key
+chmod 600 .ssh_key
+
+sshcmd='ssh -o ExitOnForwardFailure=yes -i .ssh_key -l admin'
+case "$RANCH_ENDPOINT" in
+  *huevosbuenos.com*)
+    export RANCH_ENDPOINT="https://ranch-api-staging.internal.huevosbuenos.com:8005"
+    $sshcmd -L 8005:ranch-api-staging.internal.huevosbuenos.com:443 jump.us-east-1.dev-aws.goodeggs.com "sleep 900" &
```
Comment (bobzoller) [inline install-ci-tools.sh:229]: does `sleep 900` imply we only ever leave `ranch` running for 15 minutes?  I don't think that's true if so --- eg we'll wait 30 minutes for a deploy to get healthy.

File: install-ci-tools.sh (hunk 2 above)
Comment (swettk) [reply install-ci-tools.sh:212]: It does indeed!  `man 2 kill` "If pid equals 0, then sig is sent to every process in the process group of the calling process."

**Comment by swettk** [comment]
@bobzoller worth a fresh look with the new approach.

File: install-ci-tools.sh [hunk 5]
```diff
@@ -200,7 +199,44 @@ EOF
...
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if [ -z "$RANCH_PROXY_SSH_KEY" ]
+then
+  echo "This will break in the near future, please set RANCH_PROXY_SSH_KEY"
+  exec $script_dir/ranch_real "$@"
+fi
+
+echo "$RANCH_PROXY_SSH_KEY" | base64 -d > .ssh_key
+chmod 600 .ssh_key
```
Comment (bobzoller) [inline install-ci-tools.sh:222]: there's a security race here... create the file with 600 first, then cat into it?

File: install-ci-tools.sh [hunk 6]
```diff
@@ -200,7 +199,44 @@ EOF
...
+# Make sure and clean up
+trap "exit" INT TERM ERR
+trap "kill 0" EXIT
+
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if [ -z "$RANCH_PROXY_SSH_KEY" ]
+then
+  echo "This will break in the near future, please set RANCH_PROXY_SSH_KEY"
+  exec $script_dir/ranch_real "$@"
```
Comment (bobzoller) [inline install-ci-tools.sh:218]: quote first param

File: install-ci-tools.sh [hunk 7]
```diff
@@ -200,7 +199,44 @@ EOF
...
+    export RANCH_ENDPOINT="https://ranch-api-staging.internal.huevosbuenos.com"
+    $sshcmd -D 8005 jump.us-east-1.dev-aws.goodeggs.com "sleep 3600" &
+    ;;
+  *)
+    export RANCH_ENDPOINT="https://ranch-api.internal.goodeggs.com"
+    $sshcmd -D 8005 jump.us-east-1.prod-aws.goodeggs.com "sleep 3600" &
+    ;;
+  esac
+sleep 1
+$script_dir/ranch_real "$@"
```
Comment (bobzoller) [inline install-ci-tools.sh:238]: quote first param

File: install-ci-tools.sh [hunk 8]
```diff
@@ -200,7 +199,44 @@ EOF
     ranch)
       if [ ! -x ranch ] || [ "$(./ranch version)" != "$version" ]; then
         rm -f ranch
-        curl -fsSL http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz | gunzip > ranch
+        curl -fsSL "http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz" | gunzip > ranch_real
+        chmod +x ranch_real
+      cat > ranch <<"EOF"
+#!/bin/bash
+
+set -eo pipefail
```
Comment (bobzoller) [inline install-ci-tools.sh:207]: don't forget `-u`

**Pull request merged**